### PR TITLE
[Swift] Link against the swift reflection library.

### PR DESCRIPTION
**Pull request by dcci**
This is the first step to integrate RemoteMirrors in lldb.

<rdar://problem/36477650>

### CODE REVIEW

**Comment by dcci** [comment]
@jrose-apple / @gottesmm do you know a better way of getting to the library in the cmake build system? 

@jimingham I need to hook this also up in the Xcode build system. What's the way of telling xcodebuild to link against a library?

**Review by adrian-prantl — APPROVED** [review]
We discussed this extensively before :-)

**Comment by jrose-apple** [comment]
You don't actually need the architecture, just the platform. I would expect you'd be able to use the CMake target name, `swiftReflection-PLATFORM`, but maybe that doesn't work.

**Comment by dcci** [comment]
@jrose-apple that doesn't seem to work. I noticed that the unittests use
```
swiftReflection${SWIFT_PRIMARY_VARIANT_SUFFIX}
```

but that doesn't work either.

**Comment by dcci** [comment]
@swift-ci please test and merge

**Comment by gottesmm** [comment]
This is the wrong way to do this. You should be using the export that you guys are already using from cmake.

**Comment by dcci** [comment]
force stopped the merge to discuss this with @gottesmm who's going to propose a better approach of handling this.

**Comment by gottesmm** [comment]
Thanks!

**Comment by gottesmm** [comment]
(for those listening in, I am going to show Davide how to do it tomorrow when I am in the office).

**Comment by dcci** [comment]
Will update this once the cmake bits land: https://github.com/apple/swift/pull/17525

**Comment by gottesmm** [comment]
CMake bits landed and everything looks good.

**Comment by dcci** [comment]
@swift-ci test and merge

**Comment by dcci** [comment]
(thanks @gottesmm for the help with the cmake bits on the swift side).

**Comment by dcci** [comment]
@gottesmm this is still broken for Linux. I'll take a closer look. Thanks for your help!

File: source/Target/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,13 @@
 include_directories(../Plugins/Process/Utility)
 
+# libswiftReflection is placed in a subdirectory which depends
+# on the host we're currently running on.
+# FIXME: There must be a better way to get at libswiftReflection.a
+string(REPLACE "-" ";" TRIPLE_LIST "${LLVM_HOST_TRIPLE}")
```
Comment (Rostepher) [inline source/Target/CMakeLists.txt:6]: The error message in the failing bot seems to indicate that `TRIPLE_LIST` is empty when the next line (`list(GET...)`) is called, but only on Linux.

Comment (dcci) [reply source/Target/CMakeLists.txt:6]: The issue is more complicated than that.
Linux doesn't really include the arch when building.

```
davide@rad:~/work/build$ find . -name "libswiftReflection*"
./Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/x86_64/libswiftReflection.a
./Ninja-ReleaseAssert/swift-linux-x86_64/lib/swift/linux/libswiftReflection.a
```

Even though I change `CmakeLists.txt` to just link `swiftReflection`, the library is not found, as it's not exported correctly.

**Comment by dcci** [comment]
This is obsolete now.

**Pull request closed without merging**